src/internal/control-client.ts

```diff
@@ -58,75 +54,85 @@ export class ControlClient {
     });
   }
 
-  public async createCache(name: string): Promise<CreateCacheResponse> {
-    this.validateCacheName(name);
+  public async createCache(name: string): Promise<CreateCache.Response> {
+    try {
+      this.validateCacheName(name);
+    } catch (err) {
+      if (err instanceof SdkError) {
+        return new CreateCache.Error(err);
+      } else if (err instanceof Error) {
+        return new CreateCache.Error(new UnknownError(err.message));
+      }
+    }
     this.logger.info(`Creating cache: ${name}`);
     const request = new control.control_client._CreateCacheRequest({
       cache_name: name,
     });
-    return await new Promise<CreateCacheResponse>((resolve, reject) => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    return await new Promise<CreateCache.Response>((resolve, reject) => {
       this.clientWrapper.getClient().CreateCache(
         request,
         {interceptors: this.interceptors},
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         (err, resp) => {
           if (err) {
             if (err.code === Status.ALREADY_EXISTS) {
-              reject(
-                new AlreadyExistsError(
-                  `cache with name: ${name} already exists`
-                )
-              );
+              resolve(new CreateCache.AlreadyExists());
             } else {
-              reject(cacheServiceErrorMapper(err));
+              resolve(new CreateCache.Error(cacheServiceErrorMapper(err)));
             }
           } else {
-            resolve(new CreateCacheResponse());
+            resolve(new CreateCache.Success());
           }
         }
       );
     });
   }
 
-  public async deleteCache(name: string): Promise<DeleteCacheResponse> {
+  public async deleteCache(name: string): Promise<DeleteCache.Response> {
+    try {
+      this.validateCacheName(name);
+    } catch (err) {
+      if (err instanceof SdkError) {
+        return new DeleteCache.Error(err);
+      } else if (err instanceof Error) {
+        return new DeleteCache.Error(new UnknownError(err.message));
+      }
```

Comment:
seen this little catch and wrap or not logic a bit now. Could maybe dry up with a basic utility so could do something like:
```js
    } catch (err) {
      return wrapErrorIfSDKError(DeleteCache.Error, err)
    }
```